### ISSUE TITLE
Option to Synchronize Modal Backdrop and Dialog Transitions

### ISSUE THREAD

**Issue by smkbd** [twbs/bootstrap#40873]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed feature requests
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Proposal

Currently, the Bootstrap modal with the `fade `option introduces a slight delay between the appearance of the backdrop and the modal dialog. When opening a modal with a `1s` fade transition, the backdrop transition starts first and takes `1s`, and once it completes, the modal dialog starts appearing with another `1s` transition. This results in a total delay of `2s` before the dialog is fully visible. Similarly, when closing the modal, the reverse happens: the modal dialog disappears first, followed by the backdrop.

![default-behavior](https://github.com/user-attachments/assets/32ef2e68-25da-4949-ae3d-ddb741569510)

## Proposed Feature
I propose adding an option to allow users to control whether the backdrop and modal dialog transitions are staggered (the current behavior) or occur simultaneously. This feature would give developers the flexibility to choose between the default behavior or having both the backdrop and dialog appear/disappear at the same time, enhancing the control over modal transition timing.

![proposed-feature](https://github.com/user-attachments/assets/0b47f53a-af0e-46b1-9377-07602df882a8)

## Technical Details
The current behavior is implemented in the Modal class (`js/src/modal.js`), where the show() method of the _backdrop class is called before the modal dialog itself is displayed, with the dialog appearance controlled via a callback function passed to _backdrop.show().

    // js/src/modal.js
    show(relatedTarget) {
	    
	  ...
	  
      this._backdrop.show(() => this._showElement(relatedTarget));
    }

**Backdrop Appearance**: When modal.show() is called, the _backdrop.show() method is executed first, and a callback is provided to begin showing the modal dialog only after the backdrop has completed its transition.

**Suggested Modification**: Introduce a new configuration option (e.g., `syncBackdropModal`) in the Modal class. When this option is true, the backdrop and modal dialog would transition simultaneously. This could be implemented by starting both the backdrop and modal dialog transitions together in the show() method without waiting for the callback from the backdrop transition.

**Closing Behavior**: For modal closing, a similar approach should be applied to allow simultaneous transitions for both the backdrop and modal dialog. The closing behavior currently relies on the reverse process, but further investigation is needed to confirm the exact sequence in which the modal dialog and backdrop are hidden.

### Motivation and context

If the proposed feature is implemented, it will add the following advantages-

1. Offers developers more control over the visual flow of modal dialogs.
2. Improves the UX in cases where simultaneous transitions may be more visually appealing or in line with design requirements.
3. Keeps the existing default behavior intact for backward compatibility.

**Comment by mdo** [comment]
In v6-dev the backdrop is the native `::backdrop` and transitions alongside the dialog in CSS, so the stagger is gone and timing is yours to control. Closing — please open a new issue if you need more here in v6-dev.
